cross-compile and use pixi

danielnachun opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

danielnachun committed:
cross-compile and use pixi

danielnachun committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.11.25.16.21.43

Other tools:
- conda-build 25.11.0
- rattler-build 0.51.0
- rattler-build-conda-compat 1.4.10

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

danielnachun committed:
update version

ruben-arts commented:
Hey @danielnachun, I've never seen the issue on osx-arm64. Do you think you would be able to fix that kind of issue?

We're looking forward to seeing this PR merged, as we're hoping to use dprint on linux-aarch64.

danielnachun committed:
use older rust

danielnachun committed:
MNT: Re-rendered with conda-smithy 3.54.1 and conda-forge-pinning 2025.12.22.22.48.19

Other tools:
- conda-build 25.11.1
- rattler-build 0.54.0
- rattler-build-conda-compat 1.4.10

danielnachun commented:
This is definitely a tricky error!  Usually downgrading the Rust compiler (or upgrading it if it was pinned to an older version) does the trick, but as you can see above that did not work.  To make matters more confusing, I have an Apple Silicon machine and I can't reproduce it locally.  So I'm not really sure what to do next.

The error itself is happening when linking the final binary, which makes me think there is some `.rlib` object that is missing from the final link command.  But I don't understand why there is no error on x86 macOS, nor why it used to work before.

I will note that we do have build artifacts for `linux-aarch64`, just not for the most recent version.  If you can use that older version of `dprint` for now then hopefully this will be resolved in a new release.
